runtime/runtime/src/lib.rs

```diff
@@ -281,138 +334,161 @@ impl Runtime {
         debug!(target: "runtime", "{}", log_str);
     }
 
-    fn parallel_validate_transactions(
-        config: &RuntimeConfig,
+    /// Processes a group of transactions (same signer_id and public_key), sorted by ascending nonce.
+    /// Validates, calculates cost, and performs checks sequentially.
+    /// If a transaction fails validation or verification, it invokes handle_invalid_transaction
+    /// and, if `RelaxedChunkValidation` is enabled, skips to the next transaction without failing
+    /// the whole group. The resulting state (`final_state`) contains `Account` and `AccessKey` after
+    /// the last successful transaction in the group to be committed to the `state_update`.
+    /// Returns a `GroupResult` containing verified transactions, final state, and signer_id.
+    fn verify_grouped_transactions(
+        &self,
+        group: &TransactionGroup,
+        apply_state: &ApplyState,
+        state_update: &TrieUpdate,
         gas_price: Balance,
-        signed_txs: impl IntoParallelIterator<Item = SignedTransaction>,
+        block_height: BlockHeight,
         current_protocol_version: ProtocolVersion,
-    ) -> Vec<(CryptoHash, Result<(ValidatedTransaction, TransactionCost), InvalidTxError>)> {
-        signed_txs
-            .into_par_iter()
-            .map(|signed_tx| {
-                (
-                    signed_tx.get_hash(),
-                    match validate_transaction(config, signed_tx, current_protocol_version) {
-                        Ok(validated_tx) => {
-                            match tx_cost(
-                                config,
-                                &validated_tx,
-                                gas_price,
-                                current_protocol_version,
-                            ) {
-                                Ok(cost) => Ok((validated_tx, cost)),
-                                Err(e) => Err(InvalidTxError::from(e)),
-                            }
-                        }
-                        Err((e, _tx)) => Err(e),
-                    },
-                )
-            })
-            .collect()
+        mut temp_state: Option<(Account, AccessKey)>,
+    ) -> Result<GroupResult, RuntimeError> {
+        let first_idx = group.indices[0];
+        let signer_id = group.signed_txs[first_idx].transaction.signer_id().clone();
+        let public_key = group.signed_txs[first_idx].transaction.public_key();
+
+        let mut verified = Vec::with_capacity(group.indices.len());
+
+        for &idx in &group.indices {
+            let validated_tx = match validate_transaction(
+                &apply_state.config,
+                group.signed_txs[idx].clone(),
```

Comment:
Because we need to clone the SignedTransaction here anyway, the definition of TransactionGroup can be simplified to just `Vec<SignedTransaction>`.  We clone the txs earlier and group them into lists as needed.  Then we do not have to clone here.